pkg/storage/watch_cache.go

```diff
@@ -35,9 +35,10 @@ import (
 )
 
 const (
-	// MaximumListWait determines how long we're willing to wait for a
-	// list if a client specified a resource version in the future.
-	MaximumListWait = 60 * time.Second
+	// blockTimeout determines how long we're willing to block the request
+	// to wait for a given resource version to be propagated to cache,
+	// before terminating request and returning BadRequest error.
```

Comment:
Is there a way to tie this to the users requested timeout?  It seems like if we can couple the timeout to users timeout we can reduce down to one knob to regulate